### PR TITLE
Create Home screen with components

### PR DESCRIPTION
**Pull request by yongmin86k**
# Create Home screen with components

- Add the background image of the location
    - Change the background image when the location changes by the picker
- Pass the token to App stack with `React-navigation`
    - Access the parameter using the custom javascript library `paramFromParent`
- Use compose to query the `event data` and `user data`
    - Create the query for the user's location as the default location
    - Create the query for the events in this week
- Create the `small event card` component for each event

image

image

### CODE REVIEW

File: client/js/components/CardEventSmall/CardEventSmall.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import {
+  Text,
+  View,
+  Image,
+  ImageBackground,
+  TouchableOpacity,
+} from 'react-native';
+import {RatingScore00} from '../RatingScore';
```
Comment (iyuvraajsingh) [inline client/js/components/CardEventSmall/CardEventSmall.js]: Please do not include numeric characters inside your component or class names.

**Review by iyuvraajsingh — CHANGES_REQUESTED** [review]
Great Work, Please do the required changes.

File: client/js/App.js [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
 import RootStackNavigator from '../js/navigation/RootStackNavigator';
 
 export default class App extends Component {
+  componentDidMount() {
+    if (Platform.OS === 'android') {
+      if (UIManager.setLayoutAnimationEnabledExperimental) {
+        UIManager.setLayoutAnimationEnabledExperimental(true);
+      }
+    }
+  }
```
Comment (siddharthparmar7) [inline client/js/App.js]: Nice work but remember not to spend too much time on making features compatible for android. Our focus for this project is iOS only.

File: client/js/components/CardEventSmall/CardEventSmall.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import styles from './styles';
+import PropTypes from 'prop-types';
+
+const CardEventSmall = ({eventInfo, index}) => {
```
Comment (siddharthparmar7) [inline client/js/components/CardEventSmall/CardEventSmall.js:14]: Please choose semantic names for your classes, variables and functions.

File: client/js/components/CardEventSmall/CardEventSmall.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        {category}
+      </Text>
+      <Text style={styles.title} ellipsizeMode="tail" numberOfLines={1}>
+        {title}
+      </Text>
+      <Text style={styles.price} ellipsizeMode="tail" numberOfLines={1}>
+        {price ? `$${price}` : 'Free'}
+      </Text>
+      <View style={styles.wrapperRating}>
+        <RatingScore00 />
```
Comment (siddharthparmar7) [inline client/js/components/CardEventSmall/CardEventSmall.js]: same as above.

File: client/js/components/EventItem/EventItem.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import {TouchableOpacity, ImageBackground, Text, Image} from 'react-native';
+
+const EventItem = () => {
+  return (
+    <TouchableOpacity
+      style={
+        index % 2 === 0
```
Comment (siddharthparmar7) [inline client/js/components/EventItem/EventItem.js]: what is index?

File: client/js/components/EventItem/EventItem.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+            height: '100%',
+          }}>
+          <Image
+            style={{
+              position: 'absolute',
+              bottom: 8,
+              right: 8,
+              width: 20,
+              height: 20,
+            }}
```
Comment (siddharthparmar7) [inline client/js/components/EventItem/EventItem.js]: Please move your styles to the styles.js file

File: client/js/screens/AuthLoading/AuthLoadingContainer.js [hunk 7]
```diff
@@ -9,7 +9,9 @@ class AuthLoadingContainer extends Component {
 
   _bootstrapAsync = async () => {
     const userToken = await getViewer();
-    this.props.navigation.navigate(userToken ? 'App' : 'Auth');
+    this.props.navigation.navigate(userToken ? 'App' : 'Auth', {
+      userToken: JSON.parse(userToken),
```
Comment (siddharthparmar7) [inline client/js/screens/AuthLoading/AuthLoadingContainer.js:13]: if your userToken is stored in the AsyncStorage than you don't need to pass it as param to App or Auth screen.

File: client/js/screens/Home/HomeContainer.js [hunk 8]
```diff
@@ -1,15 +1,68 @@
...
+    );
   }
 }
 
-export default HomeContainer;
+export default compose(
+  graphql(QUERY_USER, {
+    name: 'userInfo',
+    options: ({navigation}) => {
+      const userToken = getParamFromParent(navigation, 'userToken');
```
Comment (siddharthparmar7) [inline client/js/screens/Home/HomeContainer.js:54]: You should just call the AsyncStorage to get the userToken, not the params.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Nice work, Yongmin. Please make the changes.

File: client/js/components/CardEventSmall/CardEventSmall.js (hunk 1 above)
Comment (yongmin86k) [reply client/js/components/CardEventSmall/CardEventSmall.js]: Done

image

File: client/js/components/EventItem/EventItem.js (hunk 5 above)
Comment (yongmin86k) [reply client/js/components/EventItem/EventItem.js]: The nth of item

File: client/js/screens/AuthLoading/AuthLoadingContainer.js (hunk 7 above)
Comment (yongmin86k) [reply client/js/screens/AuthLoading/AuthLoadingContainer.js:13]: It is used to get the userToken with `compose()` function in a recursive way on `HomeContainer.js`

File: client/js/screens/Home/HomeContainer.js (hunk 8 above)
Comment (yongmin86k) [reply client/js/screens/Home/HomeContainer.js:54]: I can't do that in `compose()` function

File: client/js/components/CardEventSmall/CardEventSmall.js (hunk 4 above)
Comment (yongmin86k) [reply client/js/components/CardEventSmall/CardEventSmall.js]: Removed

File: client/js/components/CardEventSmall/CardEventSmall.js (hunk 3 above)
Comment (yongmin86k) [reply client/js/components/CardEventSmall/CardEventSmall.js:14]: It's a very semantic name for the small-card-event at home screen

image

File: client/js/App.js (hunk 2 above)
Comment (yongmin86k) [reply client/js/App.js]: removed

File: client/js/components/EventItem/EventItem.js (hunk 6 above)
Comment (yongmin86k) [reply client/js/components/EventItem/EventItem.js]: `EventItem.js` and its folder are the duplicated files.
It's never used so I removed it

**Comment by yongmin86k** [comment]
Please review and make comments if needed

**Review by iyuvraajsingh — APPROVED** [review]
Great Work!

**Pull request merged**